optimade/server/schemas.py

```diff
@@ -0,0 +1,118 @@
+from optimade.models import (
+    StructureResource,
+    ReferenceResource,
+    LinksResource,
+    EntryResource,
+)
+from .config import CONFIG
+
+
+def retrieve_queryable_properties(
+    schema: dict, queryable_properties: list, entry_provider_fields: list = None
+) -> dict:
+    """ For a given schema dictionary and a list of desired properties,
+    recursively descend into the schema and set the appropriate values
+    for the `description`, `sortable` and `unit` keys, returning the expanded
+    schema dict that includes nested schemas.
+
+    """
+    properties = {}
+    if entry_provider_fields is None:
+        entry_provider_fields = []
+
+    for name, value in schema["properties"].items():
+        if name in queryable_properties:
+            if "$ref" in value:
+                path = value["$ref"].split("/")[1:]
+                sub_schema = schema.copy()
+                while path:
+                    next_key = path.pop(0)
+                    sub_schema = sub_schema[next_key]
+                sub_queryable_properties = sub_schema["properties"].keys()
+                properties.update(
+                    retrieve_queryable_properties(
+                        sub_schema, sub_queryable_properties, entry_provider_fields
+                    )
+                )
+            else:
+                properties[name] = {"description": value.get("description", "")}
+                if "unit" in value:
+                    properties[name]["unit"] = value["unit"]
+                # All properties are sortable with the MongoDB backend.
+                # While the result for sorting lists may not be as expected, they are still sorted.
+                properties[name]["sortable"] = value.get("sortable", True)
```

Comment:
For `aiida-optimade` I have here a solution where I list the types that are sortable and then check the JSON type in the schema to see if `sortable` should be `True` or `False`.
I know this is irrelevant for MongoDB, since it can do it all, but maybe we shouldn't? Or maybe we should support a more general solution here?
However, that of course begs the question if `type` is the defining factor?